internal/gvr/resource.go

```diff
@@ -0,0 +1,32 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gvr
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+func ResourceFromMetaGroupVersion(gv metav1.GroupVersion, resource string) schema.GroupVersionResource {
```

Comment:
I think we can shorten this.

That said, I'm planning to change `gather` to use a controller client instead of a dynamic client in which case we would use GVK instead of GVR. Perhaps we can hold off on these changes for now.

```suggestion
func FromGroupVersion(gv metav1.GroupVersion, resource string) schema.GroupVersionResource {
```

Reply:
I've decided to keep the dynamic client for ease of use so this change is relevant.

Reply:
I’ve switched to `FromMetaGroupVersion` instead of `ResourceFrom…` because we’re specifically handling `meta` `GroupVersion` as opposed to `schema`; I’d rather keep the `Meta` reference in case we ever need to do something with one of the other versions.